api/src/list/list.service.ts

```diff
@@ -56,102 +57,40 @@ export class ListsService {
-    return movieIDs;
+    return await this.listDao.getLists(userId);
   }
 
-  async getSharees(listId: string, userId: string) {
-    const list = await this.listsDao.getList(listId);
-    const sharees = await this.listsDao.getSharees(listId);
-    const creator = sharees.user.find((user) => user.id === sharees.creatorId);
-    const filteredList = sharees.user.filter((user) => userId !== user.id);
-
-    const userListDetails = await Promise.all(
-      filteredList.map(async (user) => {
-        const userMovies = await this.getWatchedMovies(user.id);
-
-        const watchedMoviesOnList = list.movie
-          .filter((movie) => userMovies.includes(movie.id))
-          .map((movie) => movie.id);
-
-        return {
-          username: user.username,
-          email: user.email,
-          creator: user.username === creator?.username,
-          watched: watchedMoviesOnList,
-        };
-      }),
-    );
-
-    return userListDetails;
+  async getSharees(listId: string) {
+    return await this.listDao.getSharees(listId);
   }
 
-  async createList(body: CreateListDto, userId: string) {
-    const list = await this.listsDao.createList(body.name, userId);
+  async createList(name: string, userId: string) {
+    const { id } = await this.listDao.createList(name, userId);
 
-    await this.moviesService.createMovies(body.movie, list.id);
-
-    return await this.listsDao.getList(list.id);
+    return await this.listDao.getList(id);
   }
 
-  async cloneList(cloneList: CloneListDto, userId: string) {
-    const originalList = await this.listsDao.getList(cloneList.listId);
-
-    const clonedListData = {
-      name: cloneList.name,
-      movie: originalList.movie.map((movie) => ({
-        title: movie.title,
-        description: movie.description,
-        genre: [...movie.genre],
-        releaseDate: movie.releaseDate,
-        posterUrl: movie.posterUrl,
-        rating: movie.rating,
-        imdbId: movie.imdbId,
-      })),
-    };
-
-    const clonedList = await this.createList(
-      {
-        name: clonedListData.name,
-        movie: clonedListData.movie,
-      },
-      userId,
-    );
-
-    return clonedList;
-  }
+  async updateList(listId: string, updateListDto: UpdateListDto) {
+    const update: UpdateListData = {};
 
-  async updateListPrivacy(listId: string) {
-    return await this.listsDao.updateListPrivacy(listId);
-  }
+    if (updateListDto.name !== undefined) {
+      update.name = updateListDto.name;
+    }
 
-  async updateList(updateListDto: UpdateListDto) {
-    await this.moviesService.createMovies(
-      updateListDto.movie,
-      updateListDto.listId,
-    );
+    if (updateListDto.isPrivate !== undefined) {
+      update.isPrivate = updateListDto.isPrivate;
+    }
 
-    return await this.listsDao.updateList(updateListDto);
+    return await this.listDao.updateList(listId, update);
   }
 
   async deleteList(listId: string, userId: string) {
     const user = await this.usersService.getUser(userId);
```

Comment:
Why do we do this here? shouldn't the guards take care of ownership and authorization?